driver/pace/driver/run.py

```diff
@@ -1,292 +1,379 @@
-                stencil_factory=self._stencil_factory,
-                grid_data=self._grid_data,
-                namelist=self.physics_config,
-                active_packages=physics_packages,
-            )
-            self.physics_state = self._init_physics_state_from_dycore_state(
-                physics_packages
-            )
-            self.state_updater = UpdateAtmosphereState(
-                stencil_factory=self._stencil_factory,
-                grid_data=self._grid_data,
-                namelist=self._namelist,
-                comm=self._comm,
-                grid_info=DriverGridData.new_from_metric_terms(self._metric_terms),
-                quantity_factory=self._quantity_factory,
-            )
-
-    def _setup_factories(
-        self, rebuild: bool, validate_args: bool
-    ) -> Tuple["QuantityFactory", "StencilFactory"]:
-        stencil_config = pace.dsl.stencil.StencilConfig(
-            backend=self._backend,
-            rebuild=rebuild,
-            validate_args=validate_args,
+        physics_state = fv3gfs.physics.PhysicsState.init_zeros(
+            quantity_factory=quantity_factory, active_packages=["microphysics"]
         )
-        sizer = pace.util.SubtileGridSizer.from_tile_params(
-            nx_tile=self._namelist.npx - 1,
-            ny_tile=self._namelist.npy - 1,
-            nz=self._namelist.npz,
-            n_halo=N_HALO_DEFAULT,
-            extra_dim_lengths={},
-            layout=self._namelist.layout,
-            tile_partitioner=self._comm.partitioner.tile,
-            tile_rank=self._comm.tile.rank,
+        return DriverState(
+            dycore_state=dycore_state,
+            physics_state=physics_state,
+            metric_terms=metric_terms,
         )
 
-        grid_indexing = pace.dsl.stencil.GridIndexing.from_sizer_and_communicator(
-            sizer=sizer, cube=self._comm
+
+@dataclasses.dataclass
+class RestartConfig(InitializationConfig):
+    """
+    Configuration for restart initialization.
+    """
+
+    path: str
+
+    @property
+    def start_time(self) -> datetime:
+        return datetime(2000, 1, 1)
+
+    def get_driver_state(
+        self,
+        quantity_factory: pace.util.QuantityFactory,
+        communicator: pace.util.CubedSphereCommunicator,
+    ) -> DriverState:
+        metric_terms = pace.util.grid.MetricTerms(
+            quantity_factory=quantity_factory, communicator=communicator
         )
-        quantity_factory = QuantityFactory.from_backend(sizer, backend=self._backend)
-        stencil_factory = StencilFactory(
-            config=stencil_config,
-            grid_indexing=grid_indexing,
+        state = pace.util.open_restart(
+            dirname=self.path,
+            communicator=communicator,
+            quantity_factory=quantity_factory,
         )
-        return quantity_factory, stencil_factory
+        raise NotImplementedError()
+
+
+@dataclasses.dataclass(frozen=True)
+class DiagnosticsConfig:
+    path: str
```

Comment:
We originally had `Path(MODEL_OUT_DIR).mkdir(parents=True, exist_ok=True)` to make output folder. For here, do you have to make a directory first?

Reply:
Also, not for this PR, but would reading Diagnostics be a function in this class? Right now we have some limited plotting capabilities which read npy files.

Reply:
You don't have to make the directory itself, though you do have to make the directory you're going to put it in. This is probably safer behavior, because of the case where someone has a typo in a parent directory.

Ideally, the saved data should be readable by an external package, and we shouldn't need a custom object. For the one I wrote, the stored dataset is a zarr you can open with xarray. Do you mean something else by "reading Diagnostics"?

Reply:
Got it. We had it so that it will mkdir -p a model_output folder, but I can see that it could be bad if someone passes the wrong absolute path then a folder gets made somewhere else. So this is fine.

Oh good, I didn't realize it's as easy as just opening from xarray. No need to have a read functionality.